skxray/img_proc/mathops.py

```diff
@@ -0,0 +1,349 @@
+"""
+This module is designed to facilitate image arithmetic and logical operations
+on image data sets.
+"""
+
+import numpy as np
+import parser
+
+
+def arithmetic_basic(input_1,
+                     input_2,
+                     operation):
+    """
+    This function enables basic arithmetic for image processing and data
+    analysis. The function is capable of applying the basic arithmetic
+    operations (addition, subtraction, multiplication and division) to two
+    data set arrays, two constants, or an array and a constant.
+
+    Parameters
+    ----------
+    input_1 : {ndarray, int, float}
+        Specifies the first input data set, or constant, to be offset or
+        manipulated
+
+    input_2 : {ndarray, int, float}
+        Specifies the second data set, or constant, to be offset or manipulated
+
+    operation : string
+        addition: the addition of EITHER two images or volume data sets,
+            OR an image/data set and a value. This function is typically
+            used for offset purposes, or basic recombination of several isolated
+            materials or phases into a single segmented volume.
+        subtraction: enables the subtraction of EITHER one image or volume data
+            set from another, OR reduction of all values in an image/data set
+            by a set value. This function is typically used for offset
+            purposes, or basic isolation of objects or materials/phases in a
+            data set.
+        multiplication:
+
+        division:
+
+
+    Returns
+    -------
+    output : {ndarray, int, float}
+        Returns the resulting array or constant to the designated variable
+
+    Example
+    -------
+    result = mathops.arithmetic_basic(img_1, img_2, 'addition')
+    """
+    operation_dict = {'addition' : np.add,
+                      'subtraction' : np.subtract,
+                      'multiplication' : np.multiply,
+                      'division' : np.divide
+    }
+    if operation == 'division':
+        if type(input_2) is np.ndarray:
+            if 0 in input_2:
+                raise ValueError("This division operation will result in "
+                                 "division by zero values. Please reevaluate "
+                                 "denominator (input_2).")
+        else:
+            if float(input_2) == 0:
+                raise ValueError("This division operation will result in "
+                                 "division by a zero value. Please "
+                                 "reevaluate the denominator constant"
+                                 " (input_2).")
+
+    output = operation_dict[operation](input_1, input_2)
+    return output
+
+
+def arithmetic_custom(expression,
+                      A,
+                      B,
+                      C=None,
+                      D=None,
+                      E=None,
+                      F=None,
+                      G=None,
+                      H=None):

```

Comment:
A more generic version of this function can be written taking advantage of kwargs:
```python

def arithmetic_custom(expression, **kwargs):
    output = eval(parser.expr(expression).compile())
    return output
```